Source/SG/SG.cs

```diff
@@ -174,6 +172,9 @@ public static Waveform LoadWaveformFromTDMS(string filePath, string waveformName
                 waveform.BurstStopLocations = new int[1] { waveform.WaveformData.SampleCount - 1 };
             }
 
+            // calculating IdleDurationPresent like this also accounts for tools like wlan sfp that put in burst start and stop locations even if there is no idle time in the waveform
+            waveform.IdleDurationPresent = waveform.BurstStopLocations.First() - waveform.BurstStartLocations.First() != waveform.WaveformData.SampleCount - 1;

```

Comment:
I think the "correct" answer here is that First() and Last() work for any collection that implements IEnumerable. Therefore, if we decided to change the array to some other collection that doesn't implement an indexer we would be okay. I like First() and Last() because it reads easier.
You could have also asked, "Why use Last() instead of [Length - 1]?". My answer would be the same.

Reply:
Hindsight Last() isn't used here, but still illustrates my point 😆

Reply:
Thanks! 👍